src/AutoHashEqualsCached.jl

```diff
@@ -2,10 +2,16 @@
 
 module AutoHashEqualsCached
 
-using Rematch
-
 export @auto_hash_equals_cached, @auto_hash_equals
 
+function if_has_package(action::Function, name::String, uuid::Base.UUID)
```

Comment:
Not for now, but in case you haven't seen it: it sounds like 1.9 is introducing a mechanism for this type of thing, called ["package extensions"](https://julialang.org/blog/2023/04/julia-1.9-highlights/#package_extensions)

Reply:
Looking forward to it!

Reply:
This should also check that the version is recent enough (past a version number supplied by the caller).

That can be checked using this helper function:
```julia
pkgversion(m::Module) = VersionNumber(Pkg.TOML.parsefile(joinpath(dirname(string(first(methods(m.eval)).file)), "..", "Project.toml"))["version"])
```